Cubical/ZCohomology/RingStructure/CupProduct.agda

```diff
@@ -0,0 +1,91 @@
+{-# OPTIONS --safe --experimental-lossy-unification #-}
+module Cubical.ZCohomology.RingStructure.CupProduct where
+open import Cubical.ZCohomology.Base
+open import Cubical.ZCohomology.GroupStructure
+open import Cubical.ZCohomology.Properties
+
+open import Cubical.HITs.S1 hiding (_·_)
+open import Cubical.HITs.Sn
+open import Cubical.HITs.Susp
+open import Cubical.HITs.SetTruncation renaming (rec2 to sRec2)
+open import Cubical.HITs.Truncation renaming (rec to trRec)
+
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Pointed
+open import Cubical.Data.Int hiding (_+'_ ; +'≡+ ; _+_)
+open import Cubical.Data.Nat
+
+infixl 30 _·₀_
+infixr 35 _⌣ₖ_
+infixr 35 _⌣_
+
+--- This definition of ℕ-addition removes some unnecessary transports.
```

Comment:
Maybe make it private though?

Reply:
Or move to Data.Nat? Maybe it's actually useful elsewhere...

Reply:
I want to be able to use it in the other files here, so private isn't going to work.

It's really designed specifically for the cup product because of the definition of `coHomK`, so I feel like this is the most reasonable place to put it.

Reply:
Ok, let's leave it as is then!